apps/webapp/src/components/header/notifications.tsx

```diff
@@ -1,16 +1,24 @@
-import { useEffect, useMemo, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Popover, PopoverContent, PopoverTrigger } from '@penumbra-zone/ui';
 import { BellIcon } from '../../icons';
 import { useStream } from '../../fetchers/stream';
 import { BlockSync } from './block-sync';
-import { viewClient } from '../../clients/grpc.ts';
+
+import { StatusStreamResponse } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
 
 type NotificationState = 'sync' | 'notification' | 'none';
 
 export default function Notifications() {
   const [status, setStatus] = useState<NotificationState>('none');
-  const syncStream = useMemo(() => viewClient.statusStream({}), []);
-  const { data, error } = useStream(syncStream);
+  const [statusStream, setStatusStream] = useState<AsyncIterable<StatusStreamResponse>>();
+  const { data, error } = useStream(statusStream);
+
+  useEffect(() => {
+    void (async () => {
+      const { viewClient } = await import('../../clients/grpc');
+      setStatusStream(viewClient.statusStream({}));
+    })();
+  }, []);
```

Comment:
this should probably be a custom fetcher? i wasn't sure how to get `useStream` to behave, and i'm pretty sure `useEffect` will fire the import often than it should.

Reply:
yeag

<img width="624" alt="Screenshot 2023-12-13 at 23 28 07" src="https://github.com/penumbra-zone/web/assets/134443988/208965cf-1fc0-4414-a6f2-8a209174b863">